### PR TITLE
Automate subnet chain installation

### PR DESCRIPTION
**Pull request by coffeeavax**

### CODE REVIEW

File: avalancheup-aws/src/apply/mod.rs [hunk 1]
```diff
@@ -67,25 +117,418 @@ pub fn command() -> Command {
...
+            .with_prompt("Should we still proceed?")
+            .items(&["Yes...?", "No!!!"])
+            .default(0)
+            .interact()
+            .unwrap();
+        if selected == 1 {
+            return Ok(());
+        }
+    }
+
```
Comment (gyuho) [inline avalancheup-aws/src/apply/mod.rs:500]: Can we add a flag to enable subnet installation? (and disable by default)?

Looks like this is run by default?

File: avalancheup-aws/src/apply/mod.rs [hunk 2]
```diff
@@ -67,25 +117,418 @@ pub fn command() -> Command {
...
+                .num_args(1)
+                .value_parser(value_parser!(u64))
+                .default_value("30")
+        )
+        .arg(
+            Arg::new("CHAIN_RPC_URL")
+                .long("chain-rpc-url")
+                .help("Sets the P-chain or Avalanche RPC endpoint")
+                .required(true)
+                .num_args(1),
```
Comment (exdx) [inline avalancheup-aws/src/apply/mod.rs:156]: How would a user know the RPC URL ahead of time, if its created in the `apply` step?

File: avalancheup-aws/src/apply/mod.rs [hunk 3]
```diff
@@ -67,25 +117,418 @@ pub fn command() -> Command {
...
+        .arg(
+            Arg::new("SSM_DOCS")
+                .long("ssm-docs")
+                .help("Sets the hash map of AWS region to SSM document name for subnet and chain install (see avalanche-ops/src/aws/cfn-templates/ssm_install_subnet_chain.yaml)")
+                .required(false)
+                .value_parser(HashMapStringToStringParser {})
+                .num_args(1),
+        )
+        .arg(
+            Arg::new("TARGET_NODES")
```
Comment (exdx) [inline avalancheup-aws/src/apply/mod.rs:296]: how would the user know the target nodes if they are created in the `apply` step?

File: avalancheup-aws/src/apply/mod.rs [hunk 4]
```diff
@@ -67,25 +117,418 @@ pub fn command() -> Command {
...
+        )
+        .arg(
+            Arg::new("AVALANCHEGO_CONFIG_REMOTE_PATH")
+                .long("avalanchego-config-remote-path")
+                .help("avalanchego config remote file path")
+                .required(true)
+                .num_args(1),
+        )
+        .arg(
+            Arg::new("SSM_DOCS")
```
Comment (exdx) [inline avalancheup-aws/src/apply/mod.rs:288]: How would the user be able to provide this?

**Review by exdx — CHANGES_REQUESTED** [review]
Installing subnet-evm is a two step process:
* Bring up the subnet (`apply`)
* Install the chain (`install-subnet-chain`) 

The second part depends on the outputs of the first part. Simply putting them together in step 1 like this won't work, and provides a bad UX for the user. 

I think the way to automate this all is either via a script, or a new command called `install-subnet-and-chain` that uses the outputs from step 1 in step 2.

File: avalancheup-aws/src/apply/mod.rs (hunk 2 above)
Comment (gyuho) [reply avalancheup-aws/src/apply/mod.rs:156]: Yeah we only need a flag for spec file and everything else should be either in spec yaml file or default-spec flag

**Pull request closed without merging**